### PR TITLE
Feature - 5499 - Edit Team Form

### LINKED ISSUE #5499: ✨ (Admin) Edit team
## ✨ Feature

Describe the feature that is being requested

## 🕵️ Details

Add any additional details that could assist with the development of the new feature.

## 🎨 Design File

[Link to Figma](https://www.figma.com/file/UsrB15yicgDCLDQmx3snyl/User-Flows?node-id=0%3A1&t=HjevjsYNnRyAt9Nr-0)

## 📸 Screenshot

image

## 🙋‍♀️ Proposed Implementation

This should reuse most of the ui from #5497, with the addition of Description fields.

## 🌎 Localization

(optional) Provide any new copy along with translations available.

## ✅ Acceptance Criteria

A set of assumptions which, when tested, verify that the feature was properly implemented.

- [ ] Storybook story for this page
- [ ] Page actually available at `/teams/{id}/edit`
- [ ] Available to Admins, not other users
- [ ] Connected to API so that Team can actually be edited.

### PR DESCRIPTION
**Pull request by esizer**
🤖 Resolves #5499 

## 👋 Introduction

Adds a form the the `/admin/teams/{teamId}/edit` page.

## 🕵️ Details

This does not include the sub nav and restyled page header which will be added in #5707 

## 📋 TO DO

- [x] Add translations

## 🧪 Testing

Assist reviewers with steps they can take to test that the PR does what it says it does.

1. Build the app `npm run build`
2. Navigate to `/admin/teams/`
3. Click the pencil icon to edit a team
4. Change some text
5. Click the "Save team information" button
6. Confirm the team is updated

## 📸 Screenshot
localhost_8000_en_admin_teams_291131ab-9506-4334-b938-e7993e276187_edit

### CODE REVIEW

**Comment by vd1992** [comment]
there is a bunch of shared copy between create/view/edit team pages, so some caution ought to be exercised to ensure the wording is the same for all 3 pages

here is what I added with the view team PR

https://github.com/GCTC-NTGC/gc-digital-talent/blob/c68cc6f8b29c898b52e6391273f515e009cbdd7f/apps/web/src/lang/fr.json#L3319-L3322

https://github.com/GCTC-NTGC/gc-digital-talent/blob/c68cc6f8b29c898b52e6391273f515e009cbdd7f/apps/web/src/lang/fr.json#L5750-L5780

**Comment by vd1992** [comment]
an error message for attempting to submit a non-unique key (name field) will be needed
this would be shared between create and edit a team pages

image

**Comment by vd1992** [comment]
kebab-ification is somehow being triggered here if you click around on the inputs despite none of the fields being empty

gif

**Comment by esizer** [comment]
> an error message for attempting to submit a non-unique key (name field) will be needed
> this would be shared between create and edit a team pages

I'm assuming this would be part of another issue since it was not described in the AC

> kebab-ification is somehow being triggered here if you click around on the inputs despite none of the fields being empty

This is entirely intentional, it converts to kebab case on blur.

https://github.com/GCTC-NTGC/gc-digital-talent/blob/a7d9c45d06437ea70d049a740d4d28887022d0b2/apps/web/src/pages/Teams/UpdateTeamPage/components/NameField.tsx#L21-L26

File: apps/web/src/pages/Teams/UpdateTeamPage/components/DescriptionWordCounter.tsx [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import React from "react";
+import { useFormContext } from "react-hook-form";
+
+import WordCounter from "@common/components/WordCounter/WordCounter";
+
+const TEXT_AREA_MAX_WORDS = 400;
```
[the remark this answers is not in the document]
Comment (esizer) [reply apps/web/src/pages/Teams/UpdateTeamPage/components/DescriptionWordCounter.tsx]: Oops, that was a copy paste from the pool word counter 😅 

Fixed in [decrease team description to 200 word limit](https://github.com/GCTC-NTGC/gc-digital-talent/pull/5743/commits/f3b18a6fcc9ad368603fbf1389af1315000b8149)

**Review by brindasasi — APPROVED** [review]
LGTM , once the translations are in, this should be good to go ! 🥳

**Comment by brindasasi** [comment]
> @brindasasi Thanks for catching the word count validation. It has been fixed in [fix validation for team description fields](https://github.com/GCTC-NTGC/gc-digital-talent/pull/5743/commits/c929e251da7d646c55f8e962b233f2de85aa8f7f)

works great! Good to go 🙌 🥳

**Pull request merged**